src/mame/drivers/fmtowns.cpp

```diff
@@ -492,13 +492,15 @@ READ8_MEMBER(towns_state::towns_floppy_r)
 			{
 			case 1:
 				ret |= 0x0c;
-				if(m_flop[0]->get_device()->exists())
-					ret |= 0x03;
+				if(m_flop[0]->get_device())
+					if(m_flop[0]->get_device()->exists())
+						ret |= 0x03;
 				break;
 			case 2:
 				ret |= 0x0c;
-				if(m_flop[1]->get_device()->exists())
-					ret |= 0x03;
+				if(m_flop[1]->get_device())
+					if(m_flop[1]->get_device()->exists())
+						ret |= 0x03;
```

Comment:
The `&&` operator has short-circuit semantics.

Reply:
Thanks, changed them to a single if statement.